File: src/lib/categories.ts
export type TransactionType = "income" | "expense" | "transfer";

export const INCOME_CATEGORIES = [
  "Salary",
  "Bonus",
  "Freelance",
  "Investment",
  "Gift",
  "Refund",
  "OtherIncome",
] as const;

export const EXPENSE_CATEGORIES = [
  "FoodAndDrink",
  "Rent",
  "Entertainment",
  "Transportation",
  "Shopping",
  "Utilities",
  "Healthcare",
  "Education",
  "Travel",
  "OtherExpense",
] as const;

export const TRANSFER_CATEGORIES = [
  "AccountTransfer",
  "Savings",
  "LoanPayment",
  "OtherTransfer",
] as const;

export const CATEGORIES_BY_TYPE: Record<TransactionType, readonly string[]> = {
  income: INCOME_CATEGORIES,
  expense: EXPENSE_CATEGORIES,
  transfer: TRANSFER_CATEGORIES,
};

export const ALL_CATEGORIES = [
  ...INCOME_CATEGORIES,
  ...EXPENSE_CATEGORIES,
  ...TRANSFER_CATEGORIES,
] as const;

export type Category = (typeof ALL_CATEGORIES)[number];

export const CATEGORY_LABELS: Record<Category, string> = {
  Salary: "Salary",
  Bonus: "Bonus",
  Freelance: "Freelance",
  Investment: "Investment",
  Gift: "Gift",
  Refund: "Refund",
  OtherIncome: "Other Income",

  FoodAndDrink: "Food & Drink",
  Rent: "Rent",
  Entertainment: "Entertainment",
  Transportation: "Transportation",
  Shopping: "Shopping",
  Utilities: "Utilities",
  Healthcare: "Healthcare",
  Education: "Education",
  Travel: "Travel",
  OtherExpense: "Other Expense",

  AccountTransfer: "Account Transfer",
  Savings: "Savings",
  LoanPayment: "Loan Payment",
  OtherTransfer: "Other Transfer",
};

export function categoryLabel(category: string): string {
  return CATEGORY_LABELS[category as Category] ?? category;
}

File: Budgie-ios/Budgie/Core/Models/Categories.swift
//
//  Categories.swift
//  Budgie
//
//  Premade category lists + display labels (API uses enum names, never labels).
//

import Foundation

enum Categories {
    static let income = ["Salary", "Bonus", "Freelance", "Investment", "Gift", "Refund", "OtherIncome"]

    static let expense = ["FoodAndDrink", "Rent", "Entertainment", "Transportation", "Shopping",
                          "Utilities", "Healthcare", "Education", "Travel", "OtherExpense"]

    static let transfer = ["AccountTransfer", "Savings", "LoanPayment", "OtherTransfer"]

    static func byType(_ type: TransactionType) -> [String] {
        switch type {
        case .income: return income
        case .expense: return expense
        case .transfer: return transfer
        }
    }

    private static let labels: [String: String] = [
        "FoodAndDrink": "Food & Drink",
        "OtherIncome": "Other Income",
        "OtherExpense": "Other Expense",
        "AccountTransfer": "Account Transfer",
        "LoanPayment": "Loan Payment",
        "OtherTransfer": "Other Transfer"
    ]

    static func label(_ name: String) -> String {
        labels[name] ?? name
    }

    private static let icons: [String: String] = [
        "Salary": "briefcase",
        "Bonus": "gift",
        "Freelance": "briefcase",
        "Investment": "banknote",
        "Gift": "gift",
        "Refund": "arrow.counterclockwise",
        "OtherIncome": "ellipsis",
        "FoodAndDrink": "fork.knife",
        "Rent": "house",
        "Entertainment": "film",
        "Transportation": "car",
        "Shopping": "bag",
        "Utilities": "bolt",
        "Healthcare": "cross.case",
        "Education": "graduationcap",
        "Travel": "airplane",
        "AccountTransfer": "arrow.left.arrow.right",
        "Savings": "creditcard",
        "LoanPayment": "creditcard",
        "OtherTransfer": "ellipsis",
        "OtherExpense": "ellipsis"
    ]

    static func icon(_ name: String) -> String {
        icons[name] ?? "circle"
    }
}
